Core/src/TrackFitting/GainMatrixSmoother.cpp

```diff
@@ -16,20 +16,17 @@ Result<void> GainMatrixSmoother::calculate(
     const GetParameters& predicted, const GetCovariance& predictedCovariance,
     const GetCovariance& smoothedCovariance, const GetCovariance& jacobian,
     const Logger& logger) const {
-  static constexpr double epsilon = 1e-13;
-  auto regularization = BoundMatrix::Identity() * epsilon;
```

Comment:
this is actually coming from https://github.com/acts-project/acts/pull/2137 which I integrated here in order to avoid IVF FPEs. I should make this more clear in the description. The reason this hack was necessary is the ill formed cov matrix which has a huge time component basically making everything else compared to that 0